tests/components/vera/common.py

```diff
@@ -0,0 +1,856 @@
+    expected_current_temperature: Optional[int] = None,
+    expected_min_temp: Optional[int] = None,
+    expected_max_temp: Optional[int] = None,
+    expected_current_position: Optional[int] = None,
+) -> None:
+    """Assert various aspects of an HA state."""
+    device = get_device(device_id, data)
+    assert device, "Failed to find device with id %d" % device_id
+
+    entity_id = get_entity_id(device, platform)
+
+    state = hass.states.get(entity_id)  # type: State
+    assert state, f"Unknown entity {entity_id}"
+
+    if expected_state is not None:
+        assert state.state == expected_state, (
+            "Expected '%s' (type %s) but was '%s' (type %s) for entity '%s'"
+            % (
+                expected_state,
+                type(expected_state),
+                state.state,
+                type(state.state),
+                entity_id,
+            )
+        )
+
+    if expected_brightness is not None:
+        assert_state_attribute(state, "brightness", expected_brightness)
+
+    if expected_hs_color is not None:
+        assert_state_attribute(state, "hs_color", expected_hs_color)
+
+    if expected_hvac_mode is not None:
+        assert_state_attribute(state, "hvac_mode", expected_hvac_mode)
+
+    if expected_fan_mode is not None:
+        assert_state_attribute(state, "fan_mode", expected_fan_mode)
+
+    if expected_temperature is not None:
+        assert_state_attribute(state, "temperature", expected_temperature)
+
+    if expected_current_temperature is not None:
+        assert_state_attribute(
+            state, "current_temperature", expected_current_temperature
+        )
+
+    if expected_min_temp is not None:
+        assert_state_attribute(state, "min_temp", expected_min_temp)
+
+    if expected_max_temp is not None:
+        assert_state_attribute(state, "max_temp", expected_max_temp)
+
+    if expected_current_position is not None:
+        assert_state_attribute(state, "current_position", expected_current_position)
+
+
+def assert_state_attribute(state: State, attribute: str, expected: Any) -> None:
+    """Assert a state's attribute has a specific value."""
+    assert attribute in state.attributes, "Attribute '%s' was not found." % attribute
+
+    value = state.attributes.get(attribute)
+    assert value == expected, "Expected %s '%s', but was '%s' for entity '%s'" % (
+        attribute,
+        expected,
+        value,
+        state.entity_id,
+    )
+
+
+async def async_call_service(
+    hass: HomeAssistant,
+    data: ComponentData,
+    device_id: int,
+    platform: str,
+    service: str,
+    extra_attribs: dict = None,
+) -> None:
+    """Call an HA service for a vera device."""
+    await hass.services.async_call(
+        platform,
+        service,
+        {
+            **{"entity_id": get_entity_id(get_device(device_id, data), platform)},
+            **(extra_attribs or {}),
+        },
+    )
+
+    await hass.async_block_till_done()
```

Comment:
We can probably replace this with `blocking` keyword argument set to `True` in the service call.

Reply:
The turn_on/turn_off for switches and lights makes a call to update the state of the entity in order for the state to be updated. Blocking until done is the only method of ensuring we get the correct state for the time being.
The overall purpose of these tests is to set a baseline for upcoming changes I have in mind. So I'm trying to leave the component code as untouched as possible.

Reply:
Ok.